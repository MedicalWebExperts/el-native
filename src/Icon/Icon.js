import React from 'react';
import { string, number } from 'prop-types';
import VectorIcon from 'react-native-vector-icons/Ionicons';

const propTypes = {
  /**
   * Color of the icon. Default: Theme white color.
   */
  color: string,
  /**
   * Size of the icon. Default: 16.
   */
  size: number,
  /**
   * What icon to show, see Icon Explorer app or one of the links above. Default: heart.
   */
  name: string,
};

const defaultProps = {
  color: 'white',
  size: 16,
  name: 'ios-heart',
};

const Icon = props => <VectorIcon {...props} />;

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
