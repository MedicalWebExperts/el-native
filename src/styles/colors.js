export default {
  primary: 'lightgreen',
  blueColor: '#2196F3',
  lightgray: 'lightgray',
  white: 'white',
  red: 'red',
  textColor: '#757575',
  darkTextColor: '#757575',
};
