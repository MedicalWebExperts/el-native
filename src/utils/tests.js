import React from 'react';
import { shallow } from 'enzyme';

const isEquivalent = (a, b) => {
  const aProps = Object.getOwnPropertyNames(a);
  const bProps = Object.getOwnPropertyNames(b);

  if (aProps.length !== bProps.length) {
    return false;
  }

  for (let i = 0; i < aProps.length; i += 1) {
    const propName = aProps[i];
    if (!Object.is(a[propName], b[propName])) {
      return false;
    }
  }
  return true;
};

const shouldHaveStyles = (componentWithStyles, styles) => {
  expect(isEquivalent(componentWithStyles.props().style, styles)).toBeTruthy();
};

const shouldHaveText = (mainComponent, text) => {
  const wrapper = shallow(mainComponent);
  return expect(wrapper.contains(text)).toBeTruthy();
};

const shouldHaveComponent = (mainComponent, component) => {
  const wrapper = shallow(mainComponent);
  return expect(wrapper.find(component)).toHaveLength(1);
};

const shouldSimulateOnPress = (mainComponent) => {
  const cb = jest.fn();

  const cloned = React.cloneElement(mainComponent, {
    onPress: cb,
  });

  const wrapper = shallow(cloned);
  wrapper.simulate('press');
  expect(cb).toHaveBeenCalledTimes(1);
};

export { shouldHaveText, shouldHaveComponent, shouldSimulateOnPress, shouldHaveStyles };
