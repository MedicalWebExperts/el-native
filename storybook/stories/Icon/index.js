import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Icon } from '../../../src';

storiesOf('Icon', module)
  .addDecorator(getStory => (
    <CenterView style={{ backgroundColor: '#AFAFAF' }}>{getStory()}</CenterView>
  ))
  .add('Default', () => <Icon />)
  .add('Custom color', () => <Icon size={200} color="aquamarine" />)
  .add('Custom size', () => <Icon size={150} />)
  .add('Custom icon', () => <Icon size={200} name="ios-basketball" />)
  .add('Custom all', () => <Icon name="ios-football" size={250} color="azure" />);
