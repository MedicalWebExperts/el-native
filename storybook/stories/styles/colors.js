export default {
  primary: 'lightgreen',
  backgroundLight: '#F5FCFF',
  backgroundDark: '#AFAFAF',
  customColor: 'aquamarine',
  blueColor: '#2196F3',
  lightgray: 'lightgray',
  darkgray: 'gray',
  white: 'white',
};
