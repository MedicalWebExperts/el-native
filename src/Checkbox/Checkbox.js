import React from 'react';
import { bool, string, number, func } from 'prop-types';
import { Platform, TouchableOpacity } from 'react-native';
import Icon from '../Icon/Icon';

const propTypes = {
  /**
   * Color of the icon. Default: Theme white color.
   */
  color: string,
  /**
   * Size of the icon. Default: 16.
   */
  size: number,
  /**
   * Checkbox value Default: false.
   */
  checked: bool,
  /**
   * Function to be executed on press.
   */
  onPress: func,
};

const defaultProps = {
  color: 'grey',
  size: 32,
  checked: false,
  onPress: () => {},
};

const Checkbox = (props) => {
  const getName = checked =>
    `${Platform.OS === 'ios' ? 'ios' : 'md'}-${checked ? 'checkbox' : 'square'}-outline`;

  const iconProps = {
    name: getName(props.checked),
    size: props.size,
    color: props.color,
  };

  return (
    <TouchableOpacity onPress={props.onPress}>
      <Icon {...iconProps} />
    </TouchableOpacity>
  );
};

Checkbox.propTypes = propTypes;
Checkbox.defaultProps = defaultProps;

export default Checkbox;
