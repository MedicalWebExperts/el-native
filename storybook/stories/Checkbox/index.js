import React, { Component } from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import CenterView from '../CenterView';
import { Checkbox } from '../../../src';
import colors from '../styles/colors';

class CheckboxWrapper extends Component {
  state = { value: false };

  onPress = () => {
    this.setState({ value: !this.state.value });
  };

  render() {
    return (
      <Checkbox color={colors.customColor} checked={this.state.value} onPress={this.onPress} />
    );
  }
}

storiesOf('Checkbox', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => (
    <View>
      <Checkbox />
      <Checkbox checked />
    </View>
  ))
  .add('Custom size', () => (
    <View>
      <Checkbox size={120} />
      <Checkbox checked size={120} />
    </View>
  ))
  .add('Custom color', () => <CheckboxWrapper />);
