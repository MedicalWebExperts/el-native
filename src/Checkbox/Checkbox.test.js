import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from './Checkbox';
import Icon from '../Icon/Icon';
import colors from '../Theme/colors';
import { shouldHaveComponent, shouldSimulateOnPress } from '../utils/tests';

describe('Checkbox', () => {
  it('should render a Checkbox with cutom styles', () => {
    const wrapper = shallow(<Checkbox color={colors.primary} />);
    expect(wrapper.find('Icon').props().color).toBeTruthy();
  });

  it('should render a Checkbox with custom size', () => {
    const wrapper = shallow(<Checkbox size={30} />);
    expect(wrapper.find('Icon').props().size).toBeTruthy();
  });

  it('should render an Icon', () => {
    shouldHaveComponent(<Checkbox onPress={() => null} />, Icon);
  });

  test('should call the onPress', () => {
    shouldSimulateOnPress(<Checkbox onPress={() => null} />);
  });
});
