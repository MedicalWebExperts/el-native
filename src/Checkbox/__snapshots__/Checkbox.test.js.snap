// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshots should render a Checkbox with custom color 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FAFAFA",
          "fontSize": 32,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Checkbox Snapshots should render a Checkbox with custom size 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "lightgreen",
          "fontSize": 120,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Checkbox Snapshots should render a checked Checkbox 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "lightgreen",
          "fontSize": 32,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Checkbox Snapshots should render a simple Checkbox 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "lightgreen",
          "fontSize": 32,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
